pygmt/params/pattern.py

```diff
@@ -0,0 +1,111 @@
+"""
+The Pattern class for specifying bit and hachure patterns.
+"""
+
+import dataclasses
+
+from pygmt._typing import PathLike
+from pygmt.alias import Alias
+from pygmt.exceptions import GMTValueError
+from pygmt.params.base import BaseParam
+
+__doctest_skip__ = ["Pattern"]
+
+
+@dataclasses.dataclass(repr=False)
+class Pattern(BaseParam):
+    """
+    Class for specifying bit and hachure patterns.
+
+    This class allows users to specify predefined bit-patterns or custom 1-, 8-, or
+    24-bit image raster files to fill symbols and polygons in various PyGMT plotting
+    methods. The patterns can be customized with different resolution and different
+    foreground and background colors. The foreground and background colors can also be
+    inverted.
+
+    GMT provides 90 predefined patterns that can be used in PyGMT. The patterns are
+    numbered from 1 to 90, and shown below:
+
+    .. figure:: https://docs.generic-mapping-tools.org/6.5/_images/GMT_App_E.png
+       :alt: The 90 predefined bit-patterns provided with GMT
+       :width: 75%
+       :align: center
+
+    Parameters
+    ----------
+    pattern
+        The pattern to use. It can be specified in two forms:
+
+        - An integer in the range of 1-90, corresponding to one of 90 predefined 64x64
+          bit-patterns
+        - Name of a 1-, 8-, or 24-bit image raster file, to create customized, repeating
+          images using image raster files.
+    dpi
+        Resolution of the pattern in dots per inch (DPI) [Default is 300].
+    bgcolor/fgcolor
```

Comment:
Type hints for these not showing in the docs. Does it matter?

Reply:
I tried to make it work but was not successful.